Setting _XOPEN_SOURCE hides certain functions. Setting it to 500 hides strdup:

error: implicitly declaring library function 'strdup' with type 'char *(const char *)' [-Werror,-Wimplicit-function-declaration]

Raising it to 600 fixes that but hides getpagesize:

error: implicit declaration of function 'getpagesize' [-Werror,-Wimplicit-function-declaration]

Therefore, don't set it at all so that nothing is hidden.
--- lib/ext2fs/tdb.c.orig	2008-06-26 00:34:50.000000000 -0500
+++ lib/ext2fs/tdb.c	2024-05-16 09:53:55.000000000 -0500
@@ -39,7 +39,6 @@
 #define HAVE_UTIME_H
 #define HAVE_UTIME
 #endif
-#define _XOPEN_SOURCE 500
 
 #include <unistd.h>
 #include <stdio.h>
